src/mahoji/lib/abstracted_commands/slayerTaskCommand.ts

```diff
@@ -278,21 +281,37 @@ export async function slayerNewTaskCommand({
 			interactionReply(interaction, 'You cannot skip this task because Turael assigns it.');
 			return;
 		}
-
-		await handleMahojiConfirmation(
-			interaction,
-			`Really cancel task? This will reset your streak to 0 and give you a new ${slayerMaster.name} task.`
-		);
-		await prisma.slayerTask.update({
-			where: {
-				id: currentTask.id
-			},
-			data: {
-				skipped: true,
-				quantity_remaining: 0
-			}
-		});
-		await userStatsUpdate(user.id, { slayer_task_streak: 0 }, {});
+		if (currentTask?.slayer_master_id === 8) {
+			await handleMahojiConfirmation(
+				interaction,
+				`Really cancel task? This will reset your wilderness streak to 0 and give you a new ${slayerMaster.name} task.`
+			);
+			await prisma.slayerTask.update({
+				where: {
+					id: currentTask.id
+				},
+				data: {
+					skipped: true,
+					quantity_remaining: 0
+				}
+			});
+			await userStatsUpdate(user.id, { slayer_wildy_task_streak: 0 }, {});
+		} else {
+			await handleMahojiConfirmation(
+				interaction,
+				`Really cancel task? This will reset your streak to 0 and give you a new ${slayerMaster.name} task.`
+			);
+			await prisma.slayerTask.update({
+				where: {
+					id: currentTask.id
+				},
+				data: {
+					skipped: true,
+					quantity_remaining: 0
+				}
+			});
+			await userStatsUpdate(user.id, { slayer_task_streak: 0 }, {});
+		}
```

Comment:
could be simplified: 
```ts
const isUsingKris = Boolean(currentTask?.slayer_master_id === 8);
const taskStreakKey = isUsingKris ? 'slayer_wildy_task_streak' : 'slayer_task_streak';
const warning = `Really cancel task? This will reset your${isUsingKris ? " wilderness" : ""} streak to 0 and give you a new ${slayerMaster.name} task.`
```